### ISSUE TITLE
Трупы в игре / мертвые NPC

### ISSUE THREAD

**Issue by Kor-Angar** [dosinabox/g2nr_unofficial_update#159]
В труп  Ланса надо бы добавить нормальное оружие, сейчас у него палка.

Так же с трупов послушников по моему нельзя поднять посохи. 

По моему по поводу снятия оружия с трупов вопрос уже поднимался. 

Про кассию и рамераз я уже писал в соседнем топике. 

Напиши если не сложно подробнее о вставке трупамв игру и что там за нюанс с оружием. 

В модификации так же вставляется труп посыльного после взятия задания Нагура. Может его одеть в городскую одежду? Сейчас он в одежде фермера.

**Comment by Kor-Angar** [comment]
> Напиши если не сложно подробнее о вставке трупамв игру и что там за нюанс с оружиеем.

Хочу в своей сборке добавить труп рудокопа 😀

**Comment by dosinabox** [comment]
> В труп Ланса надо бы добавить нормальное оружие, сейчас у него палка.

Заменена на кинжал.

> Так же с трупов послушников по моему нельзя поднять посохи.

Они вообще без оружия. По-моему, это не неправильно.

> Напиши если не сложно подробнее о вставке трупамв игру и что там за нюанс с оружием.

С живых людей оружие снимается только если они находятся в области видимости ГГ. Да и то не всегда, на это может повлиять анимация. По этой причине - если в игру вставляется сразу труп, то у него не должно быть экипированного оружия EquipItem(self,***) - только CreateInvItems(self,***,1).

> Про кассию и рамераз я уже писал в соседнем топике.

Возможно, получится - удалять экипированные шпаги (Npc_RemoveInvItem) и сразу добавлять новые, которые не будут экипированы (CreateInvItem). Буду пробовать.

**Comment by dosinabox** [comment]
Похоже, работает)

```
func void B_KillThievesGuild()
{
	if((Andre_FoundThieves_KilledByMilitia == FALSE) && (Andre_FoundThieves_Reported == TRUE) && (Andre_FoundThieves_Reported_Day <= (Wld_GetDay() - 2)))
	{
		if(!Npc_IsDead(Cassia) || !Npc_IsDead(Jesper) || !Npc_IsDead(Ramirez))
		{
			B_KillNpc(VLK_447_Cassia);
			B_KillNpc(VLK_446_Jesper);
			B_KillNpc(VLK_445_Ramirez);
			if(Npc_HasItems(Cassia,ItMw_Meisterdegen))
			{
				Npc_RemoveInvItem(Cassia,ItMw_Meisterdegen);
				CreateInvItem(Cassia,ItMw_Meisterdegen);
			};
			if(Npc_HasItems(Jesper,ItMw_1h_Sld_Sword))
			{
				Npc_RemoveInvItem(Jesper,ItMw_1h_Sld_Sword);
				CreateInvItem(Jesper,ItMw_1h_Sld_Sword);
			};
			if(Npc_HasItems(Ramirez,ItMw_Meisterdegen))
			{
				Npc_RemoveInvItem(Ramirez,ItMw_Meisterdegen);
				CreateInvItem(Ramirez,ItMw_Meisterdegen);
			};
			Andre_FoundThieves_KilledByMilitia = TRUE;
		};
	};
};
```

**Comment by dosinabox** [comment]
![image](https://user-images.githubusercontent.com/39790055/72921399-3a24d780-3d5c-11ea-8b72-6dde5d9630ef.png)

**Comment by Kor-Angar** [comment]
Здорово. Надо тогда посмотреть трупы из игры. В 4 главе трупы ОнД появляются. Плюс труп паладина квестого в 4 главе. Во второй трупы ментов именных в Минентале.

**Comment by Kor-Angar** [comment]
В труп Ланса надо бы добавить нормальное оружие, сейчас у него палка

Заменена на кинжал.Может что то приличное положить? 😀 А то как то с Кинжал ом в Ярик.. Будто это не ланс, а стример 🙃Например ItMw_Schwert4 или ItMw_Schwert1

**Comment by dosinabox** [comment]
> Может что то приличное положить?

По деньгам много получается)

> В 4 главе трупы ОнД появляются. Плюс труп паладина квестого в 4 главе. Во второй трупы ментов именных в Минентале.

Кого-то уже исправляли. Много кого. Если кто-то остался с недоступным оружием, то нужно будет вручную прописывать удаление и создание нужного оружия. Пиши сюда всех кого найдешь.

**Comment by Kor-Angar** [comment]
> Пиши сюда всех кого найдешь.

Паладины 2 глава Миненталь. Два возле Диего. еще два в Шахте с ползунами. там же трупы рудокопов. 
и трупы ментов(ополченцев) по дороге от старой шахте к хижине Кавалорна.

**Comment by dosinabox** [comment]
> Паладины 2 глава Миненталь. Два возле Диего. еще два в Шахте с ползунами. там же трупы рудокопов.
> и трупы ментов(ополченцев) по дороге от старой шахте к хижине Кавалорна.

Экипировано ли у них оружие? Потому что если судить по скриптам, то оружия у них вообще не должно быть.

**Comment by Kor-Angar** [comment]
Нет, не экипировано. Помоему 😀 во всяком случаем точно не двуручное, его бы я заметил 😀3:23, 6 февраля 2020 г., dosinabox <notifications@github.com>:
Паладины 2 глава Миненталь. Два возле Диего. еще два в Шахте с ползунами. там же трупы рудокопов.
и трупы ментов(ополченцев) по дороге от старой шахте к хижине Кавалорна.

Экипировано ли у них оружие? Потому что если судить по скриптам, то оружия у них вообще не должно быть.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
-- Отправлено из мобильного приложения Яндекс.Почты

**Comment by Kor-Angar** [comment]
Раньше в 4 главе у ОНД был виден меч на трупе. сейчас на спине меча нет, а внутри есть.

**Comment by dosinabox** [comment]
Если оружие на спине или на поясе, то снять его нельзя.

**Comment by Kor-Angar** [comment]
может тогда его проще рядом с трупом кидать?

**Comment by dosinabox** [comment]
Тогда будет висеть в воздухе, физика не работает при вставке.

**Comment by Kor-Angar** [comment]
а вот этот вставляется как то особым образом ?     Wld_InsertItem(ItMW_Addon_Keule_2h_01,"FP_ITEM_BANDITSCAMP_16");

**Comment by dosinabox** [comment]
Под них специальные точки (фрипоинты), низко к земле. А люди и монстры вставляются на вейпоинты, они могут быть на любой высоте и повернуты как угодно.

**Comment by Kor-Angar** [comment]
> Если оружие на спине или на поясе, то снять его нельзя.

так ты же починила это у Кассии и Рамиреза? или нет

***

Арбалет Энгрома на трупе Энгрома. 4 глава, Миненталь

**Comment by dosinabox** [comment]
У Кассии и Рамиреза должно работать снятие при смерти. Не сработало?

**Comment by Kor-Angar** [comment]
когда я проверял работало. а у них они на трупе перестают весеть?

**Comment by dosinabox** [comment]
да, перестают

**Comment by Kor-Angar** [comment]
Ясно.

**Comment by dosinabox** [comment]
+ грубый короткий меч у посыльного

**Comment by Kor-Angar** [comment]
ElderGamer реализовали фичу с оружием. он вставляет живого НПС и убивает его через расписание, таким образом появляется труп и оружие на земле рядом с ним.

> Самоубиение происходит в состоянии ZS_SittingDuck. Это состояние вызывается из персонального расписания Стражей. Но можно вызывать его прямо из инстанции через

```
func void ZS_SittingDuck()
{
	if(Npc_GetDistToNpc(self,hero) > 2000)
	{
		if((self.fight_tactic == FAI_HUMAN_RANGED) && Npc_HasEquippedRangedWeapon(self))
		{
			AI_ReadyRangedWeapon(self);
		}
		else if(Npc_HasEquippedMeleeWeapon(self))
		{
			AI_ReadyMeleeWeapon(self);
		};
	};
	self.aivar[AIV_INVINCIBLE] = 1;
};

func int ZS_SittingDuck_Loop()
{
	if(self.attribute[ATR_HITPOINTS] < 1)
	{
		AI_Wait(self,1);
		return LOOP_CONTINUE;
	};
	
	Npc_ChangeAttribute(self,ATR_HITPOINTS,-self.attribute[ATR_HITPOINTS_MAX]);
	
//	Npc_RemoveInvItems(self,ItAmArrow,Npc_HasItems(self,ItAmArrow));	// Стрелы
//	Npc_RemoveInvItems(self,ItAmBolt,Npc_HasItems(self,ItAmBolt));		// Болты
//	B_RemoveInvItemsByCat_01(self,INV_POTION);							// Категория "Эликсиры"
	
	return LOOP_CONTINUE;
};

func void ZS_SittingDuck_End()
{
	
};
```

это можно использовать для мертвых ОнД и Паладинов в Минентале.

**Comment by Kor-Angar** [comment]
> это можно использовать для мертвых ОнД и Паладинов в Минентале.

послушников в 3 главе

**Comment by Kor-Angar** [comment]
* паладины 2 глава
* послушники 3 глава
* ОнД 4 глава

**Comment by dosinabox** [comment]
> * паладины 2 глава
> * послушники 3 глава
> * ОнД 4 глава

Разве они все не исправлены раньше?

**Comment by Kor-Angar** [comment]
По моему мы как обычно о разном. Я про оружие которое есть у трупа и о том как его вставить в мир на землю рядом с трупом, что бы герой мог его поднять. А ты о чём?

**Comment by Kor-Angar** [comment]
вот тут видно https://youtu.be/_EIvzD03Z_I?t=11679

**Comment by dosinabox** [comment]
теперь понятно
